mesonbuild/modules/pkgconfig.py

```diff
@@ -276,16 +287,15 @@ def generate_libs_flags(libs):
                         install_dir = l.get_custom_install_dir()[0]
                         if install_dir is False:
                             continue
+                        if uninstalled:
+                            path = os.path.dirname(state.backend.get_target_filename_abs(l))
+                            Lflag = '-L%s' % self._escape(path)
```

Comment:
Can we get away with just using absolute paths to the library in the uninstalled case? I think in the c and c++ case it should really work and avoid polluting the link command line with directories that might contain internal files that would not have been installed.

Reply:
It's a good idea, so I gave it a try but the unit test didn't pass, I'm not sure why.

The generated `libfoo-uninstalled.pc` contains `Libs: /tmp/tmpuhj3659h/libfoo.so.1.0`. That file exists. But when I run the test app that links on that lib it says: `/tmp/tmpmvxgab95/test1: error while loading shared libraries: libfoo.so.1: cannot open shared object file: No such file or directory`.

Looks like we don't support having full path in pkg-config files. Probably we keep only the basename but don't add the rpath, or something like that?

Reply:
I've just tried with my meson wrapper to make sure there are no regressions. But it still works.
One difference i see is that my wrapper generates 
```
Libs: /home/.../toolkit/_build/src/libtuiwidgets.so
```

So it uses the .so name going through the symlinks. But i can't really imagine why that should make a difference on linux.

Reply:
It's the rpath system that doesn't work with versioned soname. See the difference in the command line ninja executes:

```
[7/10] ccache cc  -o test1 'test1@exe/main.c.o' -Wl,--no-undefined -Wl,--as-needed -Wl,--start-group /tmp/tmpogah5u7m/libfoo.so.1.0 -Wl,--end-group  
```
and
```
[7/10] ccache cc  -o test1 'test1@exe/main.c.o' -Wl,--no-undefined -Wl,--as-needed -Wl,--start-group /tmp/tmpytj5nrxu/libfoo.so -Wl,--end-group -Wl,-rpath,/tmp/tmpytj5nrxu -Wl,-rpath-link,/tmp/tmpytj5nrxu  
```

Reply:
I think it's a bug in rpaths_for_bundled_shared_libraries():
```
            if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
                continue
```

The extension of `libfoo.so.1` is `.1`...

Reply:
Voilà, added a commit to fix that bug.